lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -448,18 +448,32 @@ Error Provisioner::provision(DAGListTy &networks, Module &module,
                 weightName)
                 .str());
       }
+      // Convert the weight if needed.
+      auto newTy = PH->getType();
+      auto weight = loader->getTensor();
+      if (!weight->getType().isEqual(newTy)) {
```

Comment:
This should only be the case if it's a static PH right? Maybe add an assert here?